pywinauto/controls/uia_controls.py

```diff
@@ -926,9 +928,88 @@ def texts(self):
     #----------------------------------------------------------------
     def button_count(self):
         """Return a number of buttons on the ToolBar"""
-        return len(self.children())
+        children_list = self.children()
+        if not children_list and self.element_info.handle is not None:
+            btn_count = common_controls.ToolbarWrapper(self.element_info.handle).button_count()
+            return btn_count
+        else:
+            return len(children_list)
 
     # ----------------------------------------------------------------
+    def _menu_item_exists(self, btn_x_coord, btn_y_coord):
+        # Check that element is exist
+        button_elem_info = UIAElementInfo.from_point(btn_x_coord, btn_y_coord)
+        return button_elem_info.control_type == 'MenuItem'
+
+    def _up_down_menu_items(self, btn_x_coord, btn_y_coord, side='right'):
+        # Check items up, down, left and right from current item
+        first_pos_y = btn_y_coord
+        elem_y_pos = btn_y_coord
+        menu_elem_list = []
+        direction = 0
+
+        while direction <= 1:
+            button_elem_info = UIAElementInfo.from_point(btn_x_coord, elem_y_pos)
+
+            if not direction and button_elem_info.control_type == 'MenuItem':
+                elem_height = button_elem_info.rectangle.height()
+                elem_width = button_elem_info.rectangle.width()
+                right_x = btn_x_coord + elem_width
+                left_x = btn_x_coord - elem_width
+                menu_elem_list.append(button_elem_info)
+
+                if self._menu_item_exists(right_x, elem_y_pos) and side == 'right':
+                    # Check right side from the item
+                    menu_elem_list += self._up_down_menu_items(right_x, elem_y_pos, side='right')
+                elif self._menu_item_exists(left_x, elem_y_pos):
+                    # Check left side from the item
+                    menu_elem_list += self._up_down_menu_items(left_x, elem_y_pos, side='left')
+
+                elem_y_pos += elem_height
+
+            elif not direction and button_elem_info.control_type != 'MenuItem':
+                direction = 1
+                elem_y_pos = first_pos_y
+
+            if direction and button_elem_info.control_type == 'MenuItem':
```

Comment:
I tried to do it, but I had an endless loop. I made some changes, but repetitions are in place.

Reply:
My suggestion for this method:
```python
    def _up_down_menu_items(self, btn_x_coord, btn_y_coord, side='right'):
        """Check items up, down, left and right from current item

        * **btn_x_coord** X coordinates of item.
        * **btn_y_coord** Y coordinate of item.
        * **side** flag specifies what side should check.
        """
        first_pos_y = btn_y_coord
        elem_y_pos = btn_y_coord
        menu_elem_list = []
        direction = 'down'

        while direction is not None:
            button_elem_info = UIAElementInfo.from_point(btn_x_coord, elem_y_pos)

            if button_elem_info.control_type == 'MenuItem':
                elem_height = button_elem_info.rectangle.height()
                elem_width = button_elem_info.rectangle.width()
                right_x = btn_x_coord + elem_width
                left_x = btn_x_coord - elem_width
                menu_elem_list.append(button_elem_info)

                if UIAElementInfo.from_point(right_x, elem_y_pos).control_type == 'MenuItem' and side == 'right':
                    # Check right side from the item
                    menu_elem_list += self._up_down_menu_items(right_x, elem_y_pos, side='right')
                elif UIAElementInfo.from_point(left_x, elem_y_pos).control_type == 'MenuItem':
                    # Check left side from the item
                    menu_elem_list += self._up_down_menu_items(left_x, elem_y_pos, side='left')

                if direction == 'down':
                    elem_y_pos += elem_height
                elif direction == 'up':
                    elem_y_pos -= elem_height

            elif direction == 'down':
                direction = 'up'
                elem_y_pos = first_pos_y

            elif direction == 'up':
                direction = None

        return menu_elem_list
```

Reply:
i tried this one but it's endless loop.

Reply:
There is endless loop without these changes as well. Also algorithm should be re-written because there is too large `buttons_elem_list` returned. Also more unit tests are needed to catch such situations.

Reply:
Let's stick to few levels of unit tests:
 * unit tests for not expanded menu;
 * unit tests for 1 level expanded;
 * unit tests for 2 levels expanded.

Then some tests for moving main window to bottom and right side of the screen are necessary to cover cases with non-standard menu expansion direction.